File: Middle-logic/Models/MentalObject.js
import { ObservableObject, ObservableObjectType } from "./ObservableObject.js";

export class MentalObjectType{
    static Thought = "Thought";
    static FeelingTone = "Feeling Tone";
    static MentalState = "Mental State"
}
export class MentalObject extends ObservableObject{
    constructor(id, title, description, icon, mentalObjectType, intensity){
        super(id, title, ObservableObjectType.MentalObject, description, icon, intensity)
        this.Mental_Object_Type = mentalObjectType;
    }
    static getMentalObjectFromObject(obj){
        return new MentalObject(
            obj.Observable_Object_ID,
            obj.Title,
            obj.Discriminator,
            obj.Description,
            obj.Icon,
            obj.Mental_Object_Type,
            obj.Intensity
        );
    }
}

File: backend/Models/MentalObject.cs
using System.Reflection;
using static Arm_mo.Models.SensoryStimulus;

namespace Arm_mo.Models
{
    public enum FeelingTone
    {
        Pleasant,
        Unpleasant,
        Neutral
    }
    public enum MentalState
    {
        Clear,
        Dull,
        Agitated
    } 
    public class MentalObject: ObservableObject
    {
        public enum MentalObjects
        {
            Thought,
            MentalState,
            FeelingTone
        }
        public MentalObjects MentalObjectType { get; set; }
        public FeelingTone? FeelingToneValue { get; set; }
        public MentalState? MentalStateValue { get; set; }
        public MentalObject(string title,string description,string icon, IntensityType intensity) 
            : base(title, description, icon, intensity)
        {   


        }
        //If the created object is a feeling tone
        protected MentalObject(MentalObjects mentalObjectType,FeelingTone valueOfMentalObjectType, String title, 
            string description, string icon, IntensityType intensity) : base(title, description, icon, intensity)
        {
            this.MentalObjectType= mentalObjectType;
            this.FeelingToneValue = valueOfMentalObjectType;

        }
        //If the created object is a mental state
        protected MentalObject(MentalObjects mentalObjectType, MentalState valueOfMentalObjectType, String title,
           string description, string icon, IntensityType intensity) : base(title, description, icon, intensity)
        {
            this.MentalObjectType = mentalObjectType;
            this.MentalStateValue = valueOfMentalObjectType;

        }

    }
}
